tests/graphql/errors/test-set-headers.graphql

```diff
@@ -0,0 +1,17 @@
+#> server-sdl
+schema @server(baseURL: "https://jsonplaceholder.typicode.com", setHeaders: [{key: "🤣", value: "a"}]) {
+  query: Query
+}
+
+type User {
+  name: String
+  age: Int
+}
+
+type Query {
+  user: User @const(data: {name: "John"})
+}
+
+#> client-sdl
+type Failure
+  @error(message: "Parsing failed because of invalid HTTP header name", trace: ["schema", "@server", "setHeaders"])
```

Comment:
- [ ] Add tests for invalid Http HeaderValue
- [ ] Add tests for multiple failures ie, if multiple headers are invalid, we should be able to show all of them in errors.